include/deal.II/lac/sparse_matrix_ez.h

```diff
@@ -683,6 +683,17 @@ class SparseMatrixEZ : public Subscriptor
   const_iterator begin (const size_type r) const;
 
   /**
+   * Iterator on diagonal entry of row @p r.
+   *
+   * @warning If the current row has no diagonal element, then the returned
+   * iterator will be same as end(r), i.e., an iterator points to first element
+   * of next row or same place behind end of the entire matrix.
+   * So, do not dereference or query information to the returned iterator blindly
+   * unless you can make sure your matrix do have a diagonal element in row @p r.
+   */
+  const_iterator diagonal (const size_type r) const;
```

Comment:
I found this name from the internal code of SparseMatrixEZ and feel it's not too bad. Any way the return value is an iterator, I think it is not easily to be confused with an ```std::vector<Number>```.

A name with out ambiguity may be ``` iterator_on_diagonal_element_of_row(r)```. I think this will be  too long :-(